### ISSUE TITLE
JSCS doesnt seem to lint all files on prepush hook

### ISSUE THREAD

**Issue by TDA** [mozilla/fxa-content-server#2811]
## STR
1. Run grunt jscs on your local machine:
output should be something like this:

    ![](https://photos-5.dropbox.com/t/2/AABvCaYnDR60Xiww0DH2f3ilbq_pp4Zq3Vgddm32-UKdUg/12/392896082/png/32x32/1/1437631200/0/2/Screenshot%202015-07-22%2019.26.58.png/CNK8rLsBIAEgAiADIAQgBSAGIAcoAg/c2dnNlVShYlJracO9sXABJliLujdRzdX8GzFmzGtec0?size_mode=5)
https://www.dropbox.com/s/vbuaknzucf74xsd/Screenshot%202015-07-22%2019.26.58.png?dl=0

2. Now push some files to git and you get this output:

    ![](https://photos-5.dropbox.com/t/2/AABwmwK-FzJLVBE-zAIpNiREaaAEZlQ1QsY8CaYwB38d2A/12/392896082/png/32x32/1/1437631200/0/2/Screenshot%202015-07-22%2019.33.26.png/CNK8rLsBIAEgAiADIAQgBSAGIAcoAg/0xT_abweNKnebwRtgPaPHRzqGMmlfx-XmoawvCqdDjI?size_mode=5)
    https://www.dropbox.com/s/152j0p4t6gq1bpz/Screenshot%202015-07-22%2019.33.26.png?dl=0

## Expected Behaviour
* JSCS lints all files in the app folder

## Actual Behaviour
* JSCS lints all files when run as grunt jscs, but only 1 file when on the prepush hook.

@pdehaan

**Comment by pdehaan** [comment]
Yes sir!

So, the `git pre-push` hook should be calling this npm script: `"prepush": "grunt quicklint"`.
You can check out the `quicklint` task in the [/grunttasks/lint.js:17-19](https://github.com/mozilla/fxa-content-server/blob/164b9e76b4c0dd30d1314b3d533d082e3d13bf39/grunttasks/lint.js#L17-L19) file, which wraps all the `grunt lint` tasks w/ the [**grunt-newer**](https://github.com/tschaub/grunt-newer) module, which:

> Configure Grunt tasks to run with newer files only.
> &mdash; via https://github.com/tschaub/grunt-newer

So basically the `quicklint` Grunt task should only lint modified files (to keep the pre-commit/pre-push delays to a minimum, so they don't get annoying).

**Comment by TDA** [comment]
well, that was embarassing. closing.

**Comment by pdehaan** [comment]
Not embarrassing at all. The Git hook stuff is a bit magical and probably not documented anywhere, so it's just an obscure thing buried in a grunttasks/lint.js task.
